tests/atest/rules/misc/no_test_variable/test_var_syntax.robot

```diff
@@ -0,0 +1,15 @@
+*** Variables ***
+${amazing}    Hello world!
+
+*** Test Cases ***
+Test
+    VAR    ${foo}    bar
+    VAR    ${lorum}    ipsum    scope=LOCAL
+    VAR    ${amazing}    Hello universe    scope=TEST
+
```

Comment:
Can you add some potential errors / missing stuff? For example:

```
    VAR
    VAR    $escaped_not_allowed
    VAR    ${name}    scope=typo
```

There is also dynamic scope (scope=${scope}) but we can't do anything about such abonominations

Reply:
Added those and a bunch more weirdness. I love abusing `${{ }}` evals in these situations.